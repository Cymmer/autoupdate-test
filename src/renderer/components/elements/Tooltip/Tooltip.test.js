import React from 'react';
import { shallow } from 'enzyme';
import Tooltip from './index';
import tooltipPlacement from './constants/tooltipPlacement';

describe('RENDERING - Tooltip', () => {
  it('should render with default props', () => {
    const component = shallow(
      <Tooltip content="Test">
        <p>Test</p>
      </Tooltip>
    );
    expect(component).toMatchSnapshot();
  });

  it('should set its placement appropriately', () => {
    let component = shallow(
      <Tooltip content="Test" placement={tooltipPlacement.BOTTOM}>
        <p>Test</p>
      </Tooltip>
    );
    expect(component).toMatchSnapshot();

    component = shallow(
      <Tooltip content="Test" placement={tooltipPlacement.LEFT}>
        <p>Test</p>
      </Tooltip>
    );
    expect(component).toMatchSnapshot();

    component = shallow(
      <Tooltip content="Test" placement={tooltipPlacement.RIGHT}>
        <p>Test</p>
      </Tooltip>
    );
    expect(component).toMatchSnapshot();
  });
});
