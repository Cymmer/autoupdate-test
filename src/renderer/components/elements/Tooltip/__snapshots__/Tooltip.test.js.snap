// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Tooltip should render with default props 1`] = `
<ForwardRef(TippyWrapper)
  allowHTML={true}
  animation="fade"
  className="Tooltip"
  content="Test"
  delay={0}
  duration={300}
  placement="bottom"
  theme="light"
>
  <span
    className={null}
    tabIndex={-1}
  >
    <p>
      Test
    </p>
  </span>
</ForwardRef(TippyWrapper)>
`;

exports[`RENDERING - Tooltip should set its placement appropriately 1`] = `
<ForwardRef(TippyWrapper)
  allowHTML={true}
  animation="fade"
  className="Tooltip"
  content="Test"
  delay={0}
  duration={300}
  placement="bottom"
  theme="light"
>
  <span
    className={null}
    tabIndex={-1}
  >
    <p>
      Test
    </p>
  </span>
</ForwardRef(TippyWrapper)>
`;

exports[`RENDERING - Tooltip should set its placement appropriately 2`] = `
<ForwardRef(TippyWrapper)
  allowHTML={true}
  animation="fade"
  className="Tooltip"
  content="Test"
  delay={0}
  duration={300}
  placement="left"
  theme="light"
>
  <span
    className={null}
    tabIndex={-1}
  >
    <p>
      Test
    </p>
  </span>
</ForwardRef(TippyWrapper)>
`;

exports[`RENDERING - Tooltip should set its placement appropriately 3`] = `
<ForwardRef(TippyWrapper)
  allowHTML={true}
  animation="fade"
  className="Tooltip"
  content="Test"
  delay={0}
  duration={300}
  placement="right"
  theme="light"
>
  <span
    className={null}
    tabIndex={-1}
  >
    <p>
      Test
    </p>
  </span>
</ForwardRef(TippyWrapper)>
`;
